core/src/services/analytics/sboms/service.rs

```diff
@@ -0,0 +1,359 @@
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+use async_trait::async_trait;
+
+#[allow(unused_imports)]
+use tracing::trace;
+use platform::errors::{
+    Error as PlatformError
+};
+use platform::mongodb::{
+    Service as MongoService,
+    Store as MongoStore,
+};
+use platform::mongodb::analytics::{Analytic, Stage};
+use serde_json::json;
+use crate::entities::analytics::detail::Manifest;
+use crate::services::analytics::StorageProvider;
+
+fn get_match_stage(purl: String) -> Stage {
+
+    let purl_no_quotes = purl.replace("\"", "");
+
+    println!("TRACING: || purl || (purl_no_quotes: {}) AnalyticService::generate_report({})", purl_no_quotes, purl);
+    Stage::new(
+        json!({
+            "$match": {
+                "purl": purl_no_quotes
+            }
+        })
+    )
+}
+
+fn get_match_primaries_stage() -> Stage {
+    Stage::new(
```

Comment:
I like this thing.  It's easy and I can literally copy and paste from Compass. Why are the magic strings worrisome to you?  Are there any mitigation steps we can use to prevent any problems around magic strings so we can keep this design?

Reply:
Haha, just joking.  Serde is the API. Check it out, these are equivalent: 

1. The `Magic String` solution:
```fn get_group_purls_stage() -> Stage {
    Stage::new(
        json!({
            "$group": {
                "_id": "",
                "purls": {
                  "$push": {
                    "purl": "$purl"}
                    }
                }
            }
        )
    )
}```

2. The less magical, "Value" Enum solution:
```
fn get_group_purls_stage_equivalent() -> Stage {

    let mut purl_map = Map::new();
    purl_map.insert(
        String::from("purl"),
        Value::String(String::from("$purl")),
    );

    let mut push_map = Map::new();
    push_map.insert(
        String::from("$push"),
        Value::Object(purl_map),
    );

    let mut group_value = Map::new();
    group_value.insert(
        String::from("purls"),
        Value::Object(push_map),
    );

    group_value.insert(
        String::from("_id"),
        Value::String(String::from("")),
    );

    let mut group_map = Map::new();
    group_map.insert(
        String::from("$group"),
        Value::Object(group_value),
    );


    Stage::new(Value::Object(group_map))
}```

To be sure, I implemented the `Value` solution and ran the pipeline successfully.  So, keep in mind that we are using Serde as the API and since we are using it to command MongoDB, we are taking advantage of the bindings that were developed specifically so Serde could be easily transformed into Bson.  I'm be hard pressed to develop a _better_ API than one with 3.9k stars.